pkg/cmd/gitprovider/add.go

```diff
@@ -44,9 +48,86 @@ var GitProviderAddCmd = &cobra.Command{
 		setGitProviderConfig.Username = new(string)
 		setGitProviderConfig.Alias = new(string)
 
-		err = gitprovider_view.GitProviderCreationView(ctx, apiClient, &setGitProviderConfig, existingAliases)
-		if err != nil {
-			return err
+		if len(args) == 0 {
+			err = gitprovider_view.GitProviderCreationView(ctx, apiClient, &setGitProviderConfig, existingAliases)
+			if err != nil {
+				return err
+			}
+		} else {
+			supportedProviders := config.GetSupportedGitProviders()
+			for _, gp := range supportedProviders {
+				if gp.Id == args[0] {
+					setGitProviderConfig.ProviderId = gp.Id
+					break
+				}
+			}
+			providerId := setGitProviderConfig.ProviderId
+
+			if providerId == "" {
+				supportedProvidersStr := ""
+				for _, gp := range supportedProviders {
```

Comment:
Might be easier to do `ids := util.ArrayMap(supportedProviders, func (gp ....) { return gp.Id }` and then `strings.Join(ids, ", ")`

Reply:
done